build/templates/_converters.py.mako

```diff
@@ -158,6 +158,75 @@ def convert_repeated_capabilities_without_prefix(repeated_capability):
     return ','.join(convert_repeated_capabilities(repeated_capability, ''))
 
 
+def expand_channel_string(channel_string, all_channels_in_session):
+    '''Expands a channel_string to a list of individual channel names.
+
+    The individual channel names may or may not be fully qualified channel names as applicable for
+    the session. In other words, the individual channel names will be a subset of
+    all_channels_in_session.
+
+    Examples:
+        - expand_channel_string('1', ['0', '1', '2', '3']) --> ['1']
+        - expand_channel_string('4,1:2', ['1', '2', '4']) --> ['4', '1', '2']
+        - expand_channel_string('2:3,0', ['Dev1/0', 'Dev1/1', 'Dev1/2', 'Dev1/3'])
+            --> ['Dev1/2', 'Dev1/3', 'Dev1/0']
+        - expand_channel_string('Dev1/1', ['Dev1/0', 'Dev1/1', 'Dev1/2', 'Dev1/3'])
+            --> ['Dev1/1']
+        - expand_channel_string('4,Dev1/1:2', ['Dev1/1', 'Dev1/2', 'Dev1/4'])
+            --> ['Dev1/4', 'Dev1/1', 'Dev1/2']
+        - expand_channel_string('Dev1/4,Dev1/2,Dev1/3', ['Dev1/2', 'Dev1/3', 'Dev1/4'])
+            --> ['Dev1/4', 'Dev1/2', 'Dev1/3']
+        - expand_channel_string('Dev1/1,Dev2/2', ['Dev1/0', 'Dev1/1', 'Dev1/2', 'Dev1/3', 'Dev2/0', 'Dev2/1', 'Dev2/2', 'Dev2/3'])
+            --> ['Dev1/1', 'Dev2/2']
+        - expand_channel_string(' Dev1 / 1 : 2 , 4 ', ['Dev1/1', 'Dev1/2', 'Dev1/4'])
+            --> ['Dev1/1', 'Dev1/2', 'Dev1/4']
+        - expand_channel_string('DEV1/0-1    , Dev1/3', ['dev1/0', 'dev1/1', 'dev1/2', 'dev1/3'])
+            --> ['dev1/0', 'dev1/1', 'dev1/3']
+
+    Args:
+        channel_string (str) - refer to _convert_repeated_capabilities() for the
+            supported formats (this string is expected to be used as the index of session.channels)
+
+        all_channels_in_session (list of str) - names of all the channels in the session as returned
+            by get_channel_names()
+
+    Returns:
+        channel_names (list of str) - list of strings of the expanded channels repeated capability
+            items (essentially the channel names)
+    '''
+    channel_string = channel_string.strip()
```

Comment:
I don't think you need to mutate the user's passed parameter.
Consider removing the assignment and simply doing

```
if channel_string.strip() == '':
```

Reply:
I have updated it in my latest commit.